extensions/wikia/FlowTracking/scripts/createPageTracking.js

```diff
@@ -0,0 +1,41 @@
+define(
+	'ext.wikia.flowTracking.createPageTracking',
+	['wikia.flowTracking', 'wikia.querystring', 'mw', 'wikia.document', 'wikia.window'],
+	function (flowTrack, QueryString, mw, document, window) {
+		'use strict';
+
+		var namespaceId = mw.config.get('wgNamespaceNumber'),
+			articleId = mw.config.get('wgArticleId'),
+			title = mw.config.get('wgTitle');
+
+		function trackOnEditPageLoad(editor) {
+			var qs = new QueryString(),
+				// 'flow' is the parameter passed in the url if user has started a flow already
+				flowParam = qs.getVal('flow', false),
+				tracked = qs.getVal('tracked', false);
+
+			// Do not track if the step was tracked already or article exists
+			if (tracked
+				|| articleId !== 0
+				|| (namespaceId !== 0 && (namespaceId !== -1 && title === 'CreatePage'))) {
+				return;
+			}
+
+			if (flowParam || document.referrer) {
+				flowTrack.trackFlowStep(flowParam, {editor: editor});
+			} else if (namespaceId === 0) {
+				flowTrack.beginFlow(window.wgFlowTrackingFlows.CREATE_PAGE_DIRECT_URL, {editor: editor});
+				qs.setVal('flow', window.wgFlowTrackingFlows.CREATE_PAGE_DIRECT_URL);
+			} else if (namespaceId === -1) {
+				// TODO: direct-url to Special:CreatePage (WW-351)
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS1135)